packages/near-membrane-base/src/membrane.ts

```diff
@@ -611,6 +644,43 @@ export function createMembraneMarshall() {
             return fn;
         }
 
+        function isLiveObject(value: any): boolean {
+            if (value === null || value === undefined) {
+                return false;
+            }
+            if (typeof value === 'object') {
+                if (
+                    // Arrays are marked as live in the BoundaryProxyHandler constructor.
+                    // isArrayOrNotOrThrowForRevoked(value) ||
+                    ArrayBufferIsView(value)
+                ) {
+                    return true;
+                }
+                // Plain object detection based on ObservableMembrane
+                // https://github.com/salesforce/observable-membrane/blob/v2.0.0/src/observable-membrane.ts#L46-L47
+                const proto = ReflectGetPrototypeOf(value);
+                if (
+                    proto === ObjectProto ||
+                    proto === null ||
+                    ReflectGetPrototypeOf(proto) === null
+                ) {
+                    return true;
+                }
+            }
+            if (ReflectApply(ObjectProtoHasOwnProperty, value, [LOCKER_LIVE_VALUE_MARKER_SYMBOL])) {
+                return true;
+            }
+            if (
+                !ReflectApply(ObjectProtoHasOwnProperty, value, [
+                    LOCKER_LIVE_INSTANCE_MARKER_SYMBOL,
+                ]) &&
+                ReflectHas(value, LOCKER_LIVE_INSTANCE_MARKER_SYMBOL)
```

Comment:
I think the idea here is to get the proto, then check on the class instance symbol on the proto itself. has will walk all the way up... that seems problematic.

Reply:
Should we walk prototypes checking?  So checking prototype's of prototypes too?